### ISSUE TITLE
Update of `annotations` field in volumeClaimTemplate is not reflected in `annotations` field in existing PersistentVolumeClaims

### ISSUE THREAD

**Issue by arosh** [cybozu-go/moco#558]
**Describe the bug**
We use MOCO with [pvc-autoresizer](https://github.com/topolvm/pvc-autoresizer). This is because we want to automatically resize the volumes created by MOCO in response to lack of free space by setting the annotations defined by pvc-autoresizer in MySQLCluster's volumeClaimTemplate. These annotations allow us to configure the resizing behavior, such as thresholds. Our team has adopted a change in resizing policy, so we updated the annotations of MySQLCluster's volumeClaimTemplate to apply the new policy. We expected that the updated annotations of the volumeClaimTemplate to be reflected in the annotations of the already created PersistentVolumeClaims, but unexpectedly the annotations of PersistentVolumeClaim remained unchanged.

**Environments**
- Version: MOCO v0.16.1, [pvc-autoresizer](https://github.com/topolvm/pvc-autoresizer) v0.8.1
- OS: Flatcar Container Linux (stable)

**To Reproduce**
1. Configure annotations of volumeClaimTemplate as the follows and deploy MySQLCluster

```
resize.topolvm.io/increase: 5Gi
resize.topolvm.io/storage_limit: 100Gi
resize.topolvm.io/threshold: 5Gi
```

2. Update annotations of volumeClaimTemplate as the follows.

```
resize.topolvm.io/increase: 5Gi
resize.topolvm.io/storage_limit: 200Gi
resize.topolvm.io/threshold: 5Gi
```

3. Check `kubectl describe pvc mysql-data-moco-...`

**Expected behavior**
The updated annotations of the volumeClaimTemplate to be reflected in the annotations of the already created PersistentVolumeClaims.

**Comment by masa213f** [comment]
It's by design. Please update the metadata by any operation.

The metadata of PVCs may be updated by someone other than moco, like admission webhooks.
Because of the possibility of malfunction, MOCO will not change labels and annotations of existing PVCs.

https://github.com/cybozu-go/moco/blob/v0.16.1/docs/change-pvc-template.md
> When re-creating a StatefulSet, moco-controller does not support any operation other than volume expansion as described below. It only re-creates the StatefulSet. Therefore, changes to the label and annotation assigned to the PVC must be done by the user. This is because there is a concern that if someone other than the moco-controller is editing the PVC's metadata, there may be side effects.

**Comment by masa213f** [comment]
I recognized this issue as a feature request.

Deleting the labels and annotations from existing PVCs is difficult because MOCO cannot judge that someone created the extra metadata.

But I don't see a problem with updating the value of labels and annotations if the key of the existing PVCs and the volumeClaimTemplate are the same.

**Comment by d-kuro** [comment]
* https://github.com/cybozu-go/moco/pull/613
* https://github.com/cybozu-go/moco/pull/644

This feature is available in v0.20.0 of MOCO.

https://github.com/cybozu-go/moco/blob/v0.20.0/CHANGELOG.md#0200---2023-12-19
